blades/combo_blade.h

```diff
@@ -0,0 +1,200 @@
+  ComboBlade(WS281XBladePtr<1, blade4Pin, Color8::GRB, PowerPINS<bladePowerPin2, bladePowerPin3> >(),
+             WS281XBladePtr<1, blade3Pin, Color8::GRB, PowerPINS<bladePowerPin2, bladePowerPin3>>()),
+
+In the above example we must reduce the NUM_BLADES by 1 since the ComboBlade itself
+is the only thing that counts as an entry in the blade array.
+
+*/
+
+class ComboBladeWrapper;
+
+
+/*
+ This class is used by ComboBladeWrapper to allow intercepting the style run()
+ calls sent from the blades inside the ComboBladeWrapper.
+
+ The style of the wrapped blades will be set to an instance of this class, which
+ will then call the real style with our ComboBladeWrapper instance as the
+ active blade.
+
+ Since the same style is being executed for both blades we only actually run
+ the style calls sent by the primary blade.
+
+*/
+class ComboBladeStyleWrapper : public BladeStyle {
+public:
+  // Bladestyle implementation
+  virtual void activate() override {
+    real_style_->activate();
+  }
+  virtual void deactivate() override {
+    real_style_->deactivate();
+  }
+  virtual void run(BladeBase* blade) override;
+
+  bool NoOnOff() override {
+    return real_style_->NoOnOff();
+  }
+
+  virtual bool Charging() { return real_style_->Charging();}
+
+  bool IsHandled(HandledFeature effect) override {
+    if (real_style_)
+      return real_style_->IsHandled(effect);
+    return false;
+  }
+
+  OverDriveColor getColor(int i) override {return real_style_->getColor(i);}
+
+  int get_max_arg(int arg) override { return real_style_->get_max_arg(arg);}
+
+  void setRealStyle(BladeStyle* style){real_style_ = style;}
+
+  BladeStyle* GetRealStyle(){return real_style_;}
+
+  BladeStyle* UnSetRealStyle(){
+    BladeStyle* result = real_style_;
+    real_style_ = nullptr;
+    return result;
+  }
+
+  ComboBladeStyleWrapper(ComboBladeWrapper* comboBlade, BladeBase* primary_blade) :
+    comboBlade_(comboBlade),
+    primary_blade_(primary_blade),
+    real_style_(nullptr){
+  }
+protected:
+  ComboBladeWrapper* comboBlade_;
+  BladeBase* primary_blade_;
+  BladeStyle* real_style_;
+};
+
+/*
+ This class is used by ComboBladeWrapper to allow intercepting the style run()
+ calls sent from the blades inside the ComboBladeWrapper.
+
+This class acts as our entry in the blade array. When activated, it will
+activate the loops of the two assigned blades, pointing them at an instance of
+ComboBladeStyleWrapper. This will then be used to intercept the style run()
+calls of the indivdual blades and instead run the style against the
+ComboBladeWrapper. The ComboBladeWrapper will then send the led set calls
+to the appropriate blade based on the led number.
+
+*/
+class ComboBladeWrapper : public BladeWrapper {
+public:
+  ComboBladeWrapper(BladeBase* blade1, BladeBase* blade2):
+    blade1_num_leds_(blade1->num_leds()),
+    blade2_(blade2)
+  {
+    blade_ = blade1;
+    dummy_style_ = new ComboBladeStyleWrapper(this, blade1);
```

Comment:
ProffieOS code avoids using 'new' as much as possible to avoid memory fragmentation.
Just make "dummy_style" a ComboBladeStyleWrapper instead of a ComboBladeStyleWrapper*.